Autopilot/Inc/IMU.hpp

```diff
@@ -64,8 +63,22 @@ class ICM20602: public IMU{
 
 };
 
+
 #ifdef UNIT_TESTING
 #include "IMU_Mock.hpp"
 #endif
 
+#ifdef SIMULATION
+
+// This derived class hooks into the Simulink simulation rather than hardware
+class SimulatedIMU : public IMU
+{
+    public :
+        void Begin_Measuring(){};
+        void GetResult(IMUData_t &Data);
+    private:
+        float getNewestDataPoint(const char * fileName);
```

Comment:
If you don't mind, could you please provide some clarification about how we will use this class to simulate collecting data?

Reply:
I'd suggest taking a look at https://uwarg-docs.atlassian.net/wiki/spaces/ZP/pages/924516398/Simulation for how things work but long story short, the implementation of this class exists in https://github.com/UWARG/Simulink-Sim and every time the state machines call GetResult(), they get the data from the matlab model rather than an actual sensor.